source/lambda/apiprocessor/redaction.py

```diff
@@ -208,25 +208,20 @@ def saveDocumentRedaction(documentId, bucket, table, body):
     except Exception as e:
         return (400, "invalid json document")
 
-    # fail-fast validation
-    if len(redaction.keys()) != REDACTION_NUM_OF_KEYS:
-        return (400, "bad request, redaction has additional data that is not supported."\
-                "Please only provide uuid, headers, footers and redactedItems as top level keys")
+    # validatw that we have the expected keys
+    for x in ["uuid","headers","footers","redactedItems"]:
+        if not x in redaction: return (400, "Please only provide uuid, headers, footers and redactedItems")
```

Comment:
I think you need to switch the list being iterated over with the one being checked.
The current check will return an error if the optional values are not provided.
For example,
if redaction keys does not have any footers, then the following returns an error.


`for redactionKey in redaction.keys:
        if redactionKey not in ["uuid","headers","footers","redactedItems"]: 
            return (400, "Please only provide uuid, headers, footers and redactedItems")`

Reply:
Now I understand the misunderstanding.  According to the design ["uuid","headers","footers","redactedItems"] are not optional, they must exist and be present, even if empty.  The idea is to simplify validation and also reduce the potential for human errors as these files can be modified manually in s3.  We certainly could make headers, footers and redactedItems optional but then the front-end might need to account for the possibilities as well

Reply:
Ah I see. If there arent any optional, then I'm good.